src/services/importService.ts

```diff
@@ -232,29 +225,50 @@ export default class ImportService implements IImportService {
-                const pathToUse = v1Project.file.path.replace(/\.[^/.]+$/, "");
-                const asset = AssetService.createAssetFromFilePath(
-                `file:${pathToUse}#t=${timestamp}`);
-                const assetState = originalProject.visitedFrames.indexOf(frameInt) > -1 && frameRegions.length > 0
-                ? AssetState.Tagged : (originalProject.visitedFrames.indexOf(frameInt) > -1
-                ? AssetState.Visited : AssetState.NotVisited);
-                asset.timestamp = timestamp;
-                asset.type = AssetType.VideoFrame;
-                asset.parent = parent;
-                asset.size = asset.parent.size;
-                const populatedMetadata = await assetService.getAssetMetadata(asset).then((metadata) => {
-                    const taggedMetadata = this.addRegions(metadata, frameRegions);
-                    taggedMetadata.asset.state = assetState;
-                    taggedMetadata.asset.parent = parent;
-                    return taggedMetadata;
-                });
-                generatedAssetMetadata.push(populatedMetadata);
+            if (!frameList.hasOwnProperty) {
+                continue;
             }
+            const frameRegions = frameList[frameName];
+            const frameInt = Number(frameName);
+            const timestamp = (frameInt - 1) / Number(originalProject.framerate);
+            const pathToUse = normalizeSlashes(v1Project.file.path.replace(/\.[^/.]+$/, ""));
+            const assetState = this.getAssetState(originalProject, frameRegions, frameInt);
+            const asset = this.getAsset(parent, pathToUse, timestamp);
+            const populated = await this.getPopulatedAssetMetadata(
+                assetService, asset, assetState, frameRegions, parent);
+            generatedAssetMetadata.push(populated);
         }
         return generatedAssetMetadata;
     }
+
+    private getAsset(parent: IAsset, pathToUse: string, timestamp: number): IAsset {
+        const asset = AssetService.createAssetFromFilePath(encodeFileURI(`${pathToUse}#t=${timestamp}`));
+        asset.timestamp = timestamp;
+        asset.type = AssetType.VideoFrame;
+        asset.parent = parent;
+        asset.size = asset.parent.size;
+        return asset;
+    }
+
+    private getAssetState(originalProject: any, frameRegions: IV1Region[], frameIntOrName: number|string): AssetState {
+        return originalProject.visitedFrames.indexOf(frameIntOrName) > -1 && frameRegions.length > 0
+        ? AssetState.Tagged : (originalProject.visitedFrames.indexOf(frameIntOrName) > -1
+        ? AssetState.Visited : AssetState.NotVisited);
+    }
+
+    private async getPopulatedAssetMetadata(
+            assetService: AssetService, asset: IAsset, assetState: AssetState,
+            frameRegions: IV1Region[], parent?: IAsset, includePath?: boolean): Promise<IAssetMetadata> {
+        return await assetService.getAssetMetadata(asset).then((metadata) => {
```

Comment:
Let's not mix `async` `await` with Promise `then` callback methods.  Please standardize to async/await.